src/transformStateWithClones.js

```diff
@@ -7,7 +7,38 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = [];
+  let currentState = { ...state };
+
+  for (const action of actions) {
+    let stateCopy;
```

Comment:
The variable `stateCopy` is declared inside the loop but is not used until the switch statement. It's better to declare it inside the respective cases where it's actually used.

Reply:
> The variable `stateCopy` is declared inside the loop but is not used until the switch statement. It's better to declare it inside the respective cases where it's actually used.

You can just use the `currentState` variable and get rid of `stateCopy` at all